contracts/arbitrum/connectors/dsa-spell/main.sol

```diff
@@ -0,0 +1,79 @@
+//SPDX-License-Identifier: MIT
+pragma solidity ^0.7.0;
+pragma experimental ABIEncoderV2;
+
+/**
+ * @title DSA Spell.
+ * @dev Cast spells on DSA.
+ */
+
+// import files
+import { AccountInterface } from "../../common/interfaces.sol";
+import { Stores } from "../../common/stores.sol";
+import { Events } from "./events.sol";
+
+abstract contract DSASpellsResolver is Events, Stores {
+	/**
+	 *@dev Cast spells on DSA.
+	 *@param targetDSA target DSA to cast spells on.
+	 *@param connectors Array of connector names.
+	 *@param datas Array of connector calldatas.
+	 */
+	function castOnDSA(
+		address targetDSA,
+		string[] memory connectors,
+		bytes[] memory datas
+	)
+		external
+		payable
+		returns (string memory _eventName, bytes memory _eventParam)
+	{
+		require(instaList.accountID(targetDSA) != 0, "not-a-DSA");
+
+		AccountInterface(targetDSA).cast(connectors, datas, address(this));
+
+		_eventName = "LogCastOnDSA(address,string[],bytes[])";
+		_eventParam = abi.encode(targetDSA, connectors, datas);
+	}
+
+	/**
+	 *@dev Perform spells.
+	 *@param connectors Array of connector names.
+	 *@param datas Array of connector calldatas.
+	 */
+	function spellFactory(string[] memory connectors, bytes[] memory datas)
```

Comment:
```suggestion
	function castAny(string[] memory connectors, bytes[] memory datas)
```

Reply:
@Richa-iitr @pradyuman-verma similar to https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/any

@Richa-iitr Can you add proper comments for each function? We need to have `@dev` and `@notice`